CRM/Contribute/Selector/Search.php

```diff
@@ -402,6 +402,13 @@ public function &getRows($action, $offset, $rowCount, $sort, $output = NULL) {
         }
       }
 
+      $row['balance_due'] = CRM_Core_BAO_FinancialTrxn::getPartialPaymentWithType(
+        $result->contribution_id,
```

Comment:
So it's a max 300 extra queries per search, or 100 viewing a contribution tab on a donor with 30 or donations?